### PR TITLE
Pass Completion, Rename and Block Structure options directly instead of updating the Workspace

### PR DESCRIPTION
**Pull request by tmat**
This approach will also apply to the remaining IWorkspaceOptionsProviders once necessary refactorings are complete in Roslyn.

- [x] Update Roslyn to the latest build that contains necessary changes to External Access APIs: https://github.com/dotnet/roslyn/pull/58355
- [x] Requires fix in https://github.com/dotnet/roslyn/pull/58381

### CODE REVIEW

**Comment by tmat** [comment]
@JoeRobich @333fred PTAL

**Comment by tmat** [comment]
FYI @jasonmalinowski

**Review by 333fred — APPROVED** [review]

File: src/OmniSharp.Roslyn.CSharp/Services/Refactoring/RenameService.cs [hunk 1]
```diff
@@ -41,9 +44,14 @@ public async Task<RenameResponse> Handle(RenameRequest request)
...
                 {
+                    var options = new OmniSharpRenameOptions(
+                        RenameOverloads: _omniSharpOptions.RenameOptions.RenameOverloads,
+                        RenameInStrings: _omniSharpOptions.RenameOptions.RenameInStrings,
+                        RenameInComments: _omniSharpOptions.RenameOptions.RenameInComments);
+
                     try
                     {
-                        solution = await Renamer.RenameSymbolAsync(solution, symbol, request.RenameTo, _workspace.Options);
+                        solution = await OmniSharpRenamer.RenameSymbolAsync(solution, symbol, request.RenameTo, options, nonConflictSymbols: null, CancellationToken.None);
```
Comment (jasonmalinowski) [inline src/OmniSharp.Roslyn.CSharp/Services/Refactoring/RenameService.cs:54]: This seems particularly unfortunate that an API that was entirely public is now having to get external access versions for it -- I assume we have a plan to re-make these public in some form?

Comment (tmat) [reply src/OmniSharp.Roslyn.CSharp/Services/Refactoring/RenameService.cs:54]: Possibly at some point...

**Comment by 333fred** [comment]
@tmat looks like the version you updated to doesn't have the updated types?

**Review by JoeRobich — APPROVED** [review]

**Review by 333fred — APPROVED** [review]
LGTM, assuming tests are passing

**Comment by jasonmalinowski** [comment]
Ah, this is now blocked on https://github.com/dotnet/roslyn/pull/58940

**Comment by tmat** [comment]
@333fred @JoeRobich Any idea what might be wrong with the CI build? Seems like a loader exception. Perhaps some dependency of the new Roslyn build needs binding redirects or something?

**Comment by tmat** [comment]
@jasonmalinowski Is O# not using VSMEF?

**Comment by jasonmalinowski** [comment]
We could bump Omnisharp to use VS MEF, but given this is a regression we need to fix _anyways_ I'd say let's just wait for the fix.

**Comment by jasonmalinowski** [comment]
The fix on the Roslyn side has been merged, so @tmat we just need to bump the Roslyn version in this PR (...again...).

File: build/Packages.props [hunk 2]
```diff
@@ -7,7 +7,7 @@
         <MicrosoftTestPackageVersion>17.0.0</MicrosoftTestPackageVersion>
         <MSBuildPackageVersion>17.0.0</MSBuildPackageVersion>
         <NuGetPackageVersion>6.0.0</NuGetPackageVersion>
-        <RoslynPackageVersion>4.1.0-3.22069.8</RoslynPackageVersion>
+        <RoslynPackageVersion>4.2.0-1.22074.8</RoslynPackageVersion
```
Comment (JoeRobich) [inline build/Packages.props:10]:
```suggestion
        <RoslynPackageVersion>4.2.0-1.22074.8</RoslynPackageVersion>
```

**Pull request merged**